contrib/zcp/autosnap.lua

```diff
@@ -0,0 +1,78 @@
+-- Recursively snapshot every dataset with a given property
+--
+-- Usage: zfs program <pool> autosnap.lua [noop] [<property>] <snapshot>
+
+results = {}
+
+args = ...
+argv = args["argv"]
+usage = [[
+
+
+usage: zfs program <pool> autosnap.lua [noop] [<property>] <snapshot>
+
+	noop: performs checks only, does not take snapshots
+	property: property to check. [default=com.sun:auto-snapshot]
+	snapshot: root snapshot to create example: tank@backup
+]]
+
+if #argv == 0 then
+	error(usage)
+end
+
+noop = false
+-- 'noop' is a workaround for issue #9056
```

Comment:
It'd be nice to include the proposed fix, adding the `+` prefix to `optstring` from https://github.com/zfsonlinux/zfs/issues/9056#issuecomment-512610853 so we wouldn't need this workaround.  Applying it only to `zfs program` for now should minimize any potential compatibility problems.

Reply:
Should that be included here, or should that fix be it's own PR?